opacus/privacy_engine.py

```diff
@@ -475,6 +504,7 @@ def step(self, is_empty: bool = False):
             if self.loss_reduction == "mean":
                 noise /= batch_size
 
+            # TODO: Why do we need to generate noise on every GPU if only rank0 will use it?
```

Comment:
@alexandresablayrolles is there any reason for generating the noise on every GPU, even if only one GPU is using it? 
It seems a bit like a waste of time (especially for large networks with the CSPRNG turned on).

Reply:
Fixed.